### ISSUE TITLE
Invalid PDB symbols with SourceLink, non-deterministic build

### ISSUE THREAD

**Issue by tompazourek** [Azure/Microsoft.Azure.StackExchangeRedis#39]
# Issue description

When viewing the NuGet package (2.0.0) via **NuGet Package Explorer**, I see this:

![image](https://github.com/Azure/Microsoft.Azure.StackExchangeRedis/assets/370699/83e1ec79-48f0-4990-8354-a364c41cdcea)

I also get an error when debugging process dumps in Visual Studio:

![image](https://github.com/Azure/Microsoft.Azure.StackExchangeRedis/assets/370699/bfeb216a-359d-41d5-bca3-5f1c42edd2de)

```
Source Link data for Microsoft.Azure.StackExchangeRedis.dll (embedded) failed to parse or does not conform to the schema: {"documents":{}}
```

# Untracked sources fix

I think the untracked sources (generated AssemblyInfo) warning is only minor, and can be resolved by adding an MSBuild property:

```xml
<EmbedUntrackedSources>true</EmbedUntrackedSources>
```

# Non-deterministic paths fix for SourceLink

The bigger issue is the non-deterministic build where the source file paths in the embedded symbols are not normalized, and thus don't work with SourceLink.

As seen via ILSpy:

![image](https://github.com/Azure/Microsoft.Azure.StackExchangeRedis/assets/370699/5f7de393-88bf-43c0-96f0-f10345cbf132)

The paths shouldn't be real disk paths like `C:\...`, the "normalized" paths should start with `/_/`.

This can be achieved by passing MSBuild property:

```xml
<ContinuousIntegrationBuild>true</ContinuousIntegrationBuild>
```

But this property should only be used for non-local-development builds as it affects debugging. A good way might be to pass it as `-p:ContinuousIntegrationBuild=true` in a build pipeline, or alternatively make it conditional based on some environment variable etc:

```xml
<PropertyGroup Condition="'$(TF_BUILD)'=='true'">
  <ContinuousIntegrationBuild>true</ContinuousIntegrationBuild>
</PropertyGroup>
```

I would contribute the fix, but it seems that the build pipelines are unfortunately not open-source, so I cannot confirm it will work.

**Comment by philon-msft** [comment]
Thanks for reporting this @tompazourek. We'll work on a fix for the next release.

**Comment by philon-msft** [comment]
Apologies, I forgot to address this in the v3.0.0. We'll keep this issue active to fix in the next release.

**Comment by philon-msft** [comment]
Should be fixed by #54. Let us know if you still see any issues.
